src/overlay025/poketch_system.c

```diff
@@ -489,265 +489,262 @@ static void PoketchSystem_UnloadApp(PoketchSystem *poketchSys)
+    poketchSys->currAppSave = NULL;
 }
 
-void ov25_0225424C (PoketchSystem *poketchSys)
+void PoketchSystem_NotifyAppLoaded(PoketchSystem *poketchSys)
 {
-    GF_ASSERT(poketchSys->unk_02 == 1);
-    poketchSys->unk_02 = 2;
+    GF_ASSERT(poketchSys->appState == POKETCH_APP_STATE_INIT);
+    poketchSys->appState = POKETCH_APP_STATE_UPDATE;
 }
 
-void ov25_02254260 (PoketchSystem *poketchSys)
+void PoketchSystem_NotifyAppUnloaded(PoketchSystem *poketchSys)
 {
-    GF_ASSERT(poketchSys->unk_02 == 3);
-    poketchSys->unk_02 = 0;
+    GF_ASSERT(poketchSys->appState == POKETCH_APP_STATE_SHUTDOWN);
+    poketchSys->appState = POKETCH_APP_STATE_NONE;
 }
 
-void ov25_02254274 (UnkFuncPtr_ov25_02254274 param0, void * param1)
+void PoketchSystem_SetSaveFunction(PoketchAppSaveFunction saveFunction, void *saveData)
 {
-    PoketchSystem * v0 = ov25_02253CE0();
+    PoketchSystem *poketchSys = PoketchSystem_GetFromFieldSystem();
 
-    v0->unk_44 = param0;
-    v0->unk_48 = param1;
+    poketchSys->currAppSave = saveFunction;
+    poketchSys->appSaveData = saveData;
 }
 
-static BOOL ov25_02254284 (PoketchSystem *poketchSys)
-{
-    static const TouchScreenHitTable v0[] = {
-        {4 * 8, 12 * 8, 28 * 8, 255},
-        {12 * 8, 20 * 8, 28 * 8, 255},
-        {16, 175, 16, 207}
-    };
+static const TouchScreenHitTable sMainPoketchButtons[] = {
+    {4 * 8, 12 * 8, 28 * 8, 255},   // Top button
+    {12 * 8, 20 * 8, 28 * 8, 255},  // Bottom Button
+    {16, 175, 16, 207}              // Screen
+};
 
-    poketchSys->unk_28 = PoketchButtonManager_New(v0, NELEMS(v0), ov25_022542E4, poketchSys, 7);
+static BOOL PoketchSystem_ButtonInit(PoketchSystem *poketchSys)
+{
+    poketchSys->buttonManager = PoketchButtonManager_New(sMainPoketchButtons, NELEMS(sMainPoketchButtons), PoketchSystem_OnButtonEvent, poketchSys, 7);
 
-    if (poketchSys->unk_28 != NULL) {
-        PoketchButtonManager_SetButtonTimer(poketchSys->unk_28, 0, 0, 7);
+    if (poketchSys->buttonManager != NULL) {
+        PoketchButtonManager_SetButtonTimer(poketchSys->buttonManager, 0, 0, 7);
         poketchSys->unk_2C = 0xffffffff;
         poketchSys->unk_30 = 0xffffffff;
-        return 1;
+        return TRUE;
     }
 
-    return 0;
+    return FALSE;
 }
 
-static void ov25_022542C8 (PoketchSystem *poketchSys)
+static void PoketchSystem_ButtonShutdown(PoketchSystem *poketchSys)
 {
-    PoketchButtonManager_Free(poketchSys->unk_28);
+    PoketchButtonManager_Free(poketchSys->buttonManager);
 }
 
-static void ov25_022542D4 (PoketchSystem *poketchSys)
+static void PoketchSystem_ButtonUpdate(PoketchSystem *poketchSys)
 {
-    poketchSys->unk_08 = 0;
-    PoketchButtonManager_Update(poketchSys->unk_28);
+    poketchSys->buttonState = 0;
+    PoketchButtonManager_Update(poketchSys->buttonManager);
 }
 
-static void ov25_022542E4 (u32 param0, u32 param1, u32 param2, void * param3)
+static void PoketchSystem_OnButtonEvent(u32 buttonID, u32 buttonEvent, u32 touchEvent, void *system)
```

Comment:
```c
static void PoketchSystem_OnButtonEvent(u32 buttonID, u32 buttonEvent, u32 touchEvent, PoketchSystem *poketchSystem)
{
    if (!ov25_0225450C(poketchSystem)) {
```

Reply:
Bump

Reply:
I don't think we can do this one--or if we do, maybe in a separate PR. The function `PoketchButtonManager_New` requires this function signature, and that's called in almost every poketch app file so changing it would require changing function signatures in almost every app. This PR is already pretty big so I'm inclined to hold off for now.